### PR TITLE
Expose POST /:id/expire routes for Judgement, AssetMapping, AssetProperties

### PR DESCRIPTION
**Pull request by frenchy64**
Related https://github.com/threatgrid/iroh/issues/4293

All routes support `wait_for` and Judgement requires an extra `reason` query parameter.

Previously AssetMapping and AssetProperties used `/expire/:id` instead of `/:id/expire`. The implementation of those routes has been generalized so it can be activated by any entity via `entity-crud-routes` (assuming it doesn't have fancy requirements like extra query parameters).

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Expose POST /:id/expire routes for Judgement, AssetMapping, AssetProperties
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: src/ctia/lib/utils.clj [hunk 1]
```diff
@@ -173,3 +173,21 @@
                           service-fns)))))
           {}
           (partition-all 2 selectors)))
+
+(defn assoc-new-keys
+  "Like assoc, except only associates keys contained in m.
+  Note: for duplicate keys in kv, last wins.
+  
+  (assoc-new-keys m k v ...) is like (into (hash-map k v ...) m)
```
Comment (agzam) [inline src/ctia/lib/utils.clj:181]: Hmmm... this statement is a bit confusing:

```
(into (hash-map :foo 2) {:foo 1}) ; => {:foo 1}
(into (hash-map :foo 1) {:foo 2}) ; => {:foo 2}
(assoc-new-keys {:foo 1} :foo 2) ; => {:foo 1}
```

For data transform functions like that, maybe it's useful to attach some examples. Perhaps even something like this: https://github.com/threatgrid/ctia/blob/master/test/ctia/test_helpers/pagination.clj#L128

Comment (frenchy64) [reply src/ctia/lib/utils.clj:181]: Thanks, I'm going to ditch this for `into`.

Comment (frenchy64) [reply src/ctia/lib/utils.clj:181]: It's always the helpers that get feedback isn't it... :)

Comment (frenchy64) [reply src/ctia/lib/utils.clj]: FWIW your second example had the kv and m backwards from the description. The first and third example matched in both pattern and result.

Comment (agzam) [reply src/ctia/lib/utils.clj]: > It's always the helpers that get feedback isn't it... :)

Probably because they are small and easily REPLable.

Comment (frenchy64) [reply src/ctia/lib/utils.clj]: Easy bikesheddable! :D

**Review by ereteog — APPROVED** [review]
Nice!

File: src/ctia/entity/asset_mapping.clj [hunk 2]
```diff
@@ -164,7 +125,8 @@
      :external-id-capabilities :read-asset-mapping
      :can-aggregate?           true
      :histogram-fields         asset-mapping-histogram-fields
-     :enumerable-fields        asset-mapping-enumerable-fields})))
+     :enumerable-fields        asset-mapping-enumerable-fields
+     :can-revoke?              asset-mapping-can-revoke?}))
```
[the remark this answers is not in the document]
Comment (frenchy64) [reply src/ctia/entity/asset_mapping.clj:129]: I use these extra variables in the tests to activate the `/:id/expire` tests.

**Comment by frenchy64** [comment]
I'll merge this then test it on INT. Then I'll see what documentation we need and if all the stakeholders are happy.

**Comment by frenchy64** [comment]
Oh I forgot QA. I'll fill that out now.

**Comment by kaursundeep84** [comment]
Verified in v1.64

**Pull request merged**